### PR TITLE
fix(search): fuzzy rewrite parameter not work

### PR DESCRIPTION
**Pull request by crowod**
In the `StringFieldType.fuzzyQuery` method, the `rewriteMethod` parameter is not used for the constructor of `FuzzyQuery`. This means that passing a rewrite parameter in `fuzzy query` will not work.

### CODE REVIEW

**Comment by thecoop** [comment]
@crowod Could you add a test showing the fixed behaviour?

**Comment by crowod** [comment]
> @crowod Thanks for the contribution - could you add a test showing the fixed behaviour?

ok, I have added a test in `TextFieldTypeTests` because I found this issue when using `fuzzyQuery` with a `text` type field. Do I need to write a separate test for `StringFieldType`?

File: server/src/test/java/org/elasticsearch/index/mapper/TextFieldTypeTests.java [hunk 1]
```diff
@@ -10,18 +10,7 @@
...
-import org.apache.lucene.search.ConstantScoreQuery;
-import org.apache.lucene.search.FuzzyQuery;
-import org.apache.lucene.search.PrefixQuery;
-import org.apache.lucene.search.Query;
-import org.apache.lucene.search.RegexpQuery;
-import org.apache.lucene.search.TermInSetQuery;
-import org.apache.lucene.search.TermQuery;
-import org.apache.lucene.search.TermRangeQuery;
-import org.apache.lucene.search.WildcardQuery;
+import org.apache.lucene.search.*;
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/index/mapper/TextFieldTypeTests.java:13]: Could you run `gradlew spotlessApply` on your branch, this doesn't match our import style

Comment (crowod) [reply server/src/test/java/org/elasticsearch/index/mapper/TextFieldTypeTests.java]: > Could you run `gradlew spotlessApply` on your branch, this doesn't match our import style

I'm sorry, this is my fault. It's possible that my IDE performed some operations that I didn't anticipate and I didn't check this part carefully. Now, I have executed `gradlew spotlessApply` to adjust the relevant code style.

**Comment by mayya-sharipova** [comment]
@elasticsearchmachine test this please

**Comment by mayya-sharipova** [comment]
@elasticmachine update branch

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Review by mayya-sharipova — APPROVED** [review]
@crowod Thanks, this is a nice correction.

File: docs/changelog/97642.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pr: 97642
+summary: fix fuzzy query rewrite parameter not work
+area: Search
+type: fix
```
Comment (mayya-sharipova) [inline docs/changelog/97642.yaml:4]: @crowod We need to change here from "type:fix" to "type: bug"

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Comment by mayya-sharipova** [comment]
@elasticmachine update branch

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Comment by mayya-sharipova** [comment]
@crowod Thank you for your work, it has been merged for the 8.11 release.

**Pull request merged**